addon/services/device/layout.js

```diff
@@ -45,10 +45,12 @@ export default Service.extend({
     let w = Math.max(document.documentElement.clientWidth, window.innerWidth || 0);
     let h = Math.max(document.documentElement.clientHeight, window.innerHeight || 0);
 
-    this.setProperties({
-      width: w,
-      height: h
-    });
+    if (!this.get('isDestroyed') || !this.get('isDestroying')) {
```

Comment:
Also, lets guard right at the top of this method and return early if destroying or destroyed.